### ISSUE TITLE
Small bug on `ExoticStatistics` Decorator

### ISSUE THREAD

**Issue by bblodfon** [xoopR/distr6#273]
``` r
library(distr6)

x = Matdist$new(pdf = matrix(0.5, 2, 2, dimnames = list(NULL, 1:2)), decorators = "ExoticStatistics")

x$survival(c(1,10)) # ok
#>        1 10
#> [1,] 0.5  0
#> [2,] 0.5  0

# Next are not ok
x$survivalAntiDeriv() 
#> Results from numeric calculations are approximate only. Better results may be available.
#> Error in stats::integrate(f = function(x) abs(fun(x))^p, lower, upper): argument "lower" is missing, with no default
x$survivalAntiDeriv(lower = 0, upper = 1)
#> Results from numeric calculations are approximate only. Better results may be available.
#> Error in stats::integrate(f = function(x) abs(fun(x))^p, lower, upper): argument "lower" is missing, with no default
x$survivalPNorm(lower = 0, upper = 1)
#> Results from numeric calculations are approximate only. Better results may be available.
#> Error in stats::integrate(f = function(x) abs(fun(x))^p, lower, upper): argument "lower" is missing, with no default
x$cdfAntiDeriv()
#> Results from numeric calculations are approximate only. Better results may be available.
#> Error in stats::integrate(f = function(x) abs(fun(x))^p, lower, upper): argument "lower" is missing, with no default
x$cdfPNorm()
#> Results from numeric calculations are approximate only. Better results may be available.
#> Error in stats::integrate(f = function(x) abs(fun(x))^p, lower, upper): argument "lower" is missing, with no default

x$workingSupport()
#> {1}^n
as.numeric(x$workingSupport())
#> NULL
```
<sup>Created on 2022-04-11 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Maybe related to the parameter `range` becoming `NULL`, e.g. see: https://github.com/alan-turing-institute/distr6/blob/main/R/DistributionDecorator_ExoticStatistics.R#L151

**Comment by bblodfon** [comment]
@RaphaelS1 FYI, I found this while playing around with the `ExoticStatistics` methods on the [mlr3proba issue](https://github.com/mlr-org/mlr3proba/issues/273) we discussed.

**Comment by RaphaelS1** [comment]
Good catch (sorry for the very long wait). Bug seems to be cause by `workingSupport` and `as.numeric` not playing nice with the power set from set6.

Closed in #278 

``` r
library(distr6)
#> 
#> Attaching package: 'distr6'
#> The following object is masked from 'package:stats':
#> 
#>     qqplot
#> The following object is masked from 'package:base':
#> 
#>     truncate
x = Matdist$new(pdf = matrix(0.5, 2, 2, dimnames = list(NULL, 1:2)), decorators = "ExoticStatistics")
x$survival(c(1, 10))
#>    [,1] [,2]
#> 1   0.5  0.5
#> 10  0.0  0.0
x$survivalAntiDeriv()
#> [1] 1
x$survivalAntiDeriv(lower = 0, upper = 1)
#> [1] 0
x$survivalPNorm(lower = 0, upper = 1)
#> [1] 0
x$cdfAntiDeriv()
#> [1] 1
x$cdfPNorm()
#> [1] 0.5
x$workingSupport()
#> {1, 2}
as.numeric(x$workingSupport())
#> [1] 1 2
```

<sup>Created on 2022-08-02 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by RaphaelS1** [comment]
@bblodfon some bugs dont need to be squished https://github.com/xoopR/distr6/commit/a7c01f7166f67ebbfdc8bec4b5625ab911af85c4
